dataloader/dataloader.hpp

```diff
@@ -0,0 +1,245 @@
+   * @param dropHeader Drops the first row from CSV.
+   * @param startInputFeatures First Index which will be fed into the model as input.
+   *                           Note: Indicies are wrapped and -1 implies last
+   *                           column.
+   * @param endInputFeature Last Index which will be fed into the model as input.
+   *                        Note: Indicies are wrapped and -1 implies last
+   *                        column.
+   * @param startPredictionFeatures First Index which be predicted by the model as output.
+   *                                Note: Indicies are wrapped and -1 implies last
+   *                                column.
+   * @param endPredictionFeatures Last Index which be predicted by the model as output.
+   *                              Note: Indicies are wrapped and -1 implies last
+   *                              column.
+   * @param augmentation Vector strings of augmentations supported by mlpack.
+   * @param augmentationProbability Probability of applying augmentation to a particular cell.
+   */
+  void LoadCSV(const std::string &datasetPath,
+               const bool loadTrainData = true,
+               const bool shuffle = true,
+               const double ratio = 0.75,
+               const bool useScaler = false,
+               const bool dropHeader = false,
+               const int startInputFeatures = -1,
+               const int endInputFeatures = -1,
+               const int startPredictionFeatures = -1,
+               const int endPredictionFeatures = -1,
+               const std::vector<std::string> augmentation =
+                   std::vector<std::string>(),
+               const double augmentationProbability = 0.2);
+
+  //! Get the Training Dataset.
+  DatasetX TrainX() const { return trainX; }
+
+  //! Modify the Training Dataset.
+  DatasetX& TrainX() { return trainX; }
+
+  //! Get the Training Dataset.
+  DatasetY TrainY() const { return trainY; }
+  //! Modify the Training Dataset.
+  DatasetY& TrainY() { return trainY; }
+
+  //! Get the Test Dataset.
+  DatasetX TestX() const { return testX; }
+  //! Modify the Test Dataset.
+  DatasetX& TestX() { return testX; }
+
+  //! Get the Test Dataset.
+  DatasetY TestY() const { return testY; }
+  //! Modify the Training Dataset.
+  DatasetY& TestY() { return testY; }
+
+  //! Get the Validation Dataset.
+  DatasetX ValidX() const { return validX; }
+  //! Modify the Validation Dataset.
+  DatasetX& ValidX() { return validX; }
+
+  //! Get the Validation Dataset.
+  DatasetY ValidY() const { return validY; }
+  //! Modify the Validation Dataset.
+  DatasetY& ValidY() { return validY; }
+
+  //! Get the Scaler.
+  ScalerType Scaler() const { return scaler; }
+  //! Modify the Sclaer.
+  ScalerType& Scaler() { return scaler; }
+
+ private:
+  /**
+   * Downloads and checks hash for given dataset.
+   */
+  void DownloadDataset(const std::string& dataset)
```

Comment:
Might be nice to comment on the parameter even if it's a private function.

Reply:
Sure, Sorry I missed it.